RealTimeFaceMaskDetector/TCPServer/WindowsService.cpp

```diff
@@ -122,16 +122,18 @@ bool Service::CreateServer()
 	LOG_MSG << "CreateServer begin";
 	bool is_server_initialized = s_socket_server.InitSocketServer();
 	bool is_socked_created = s_socket_server.CreateListeningSocket();
-	bool is_listening_started = s_socket_server.StartListening();
-	bool is_connection_accepted = s_socket_server.AcceptConnection();
-	bool is_message_received = s_socket_server.ReceiveMessage();
+	bool is_listening_started;
+
+	std::thread listen_multiple_clients([&]() {s_socket_server.StartListening(is_listening_started); });
+	if (listen_multiple_clients.joinable())
+	{
+		listen_multiple_clients.join();
```

Comment:
Here if we call detach(), the server will be able to listen to multiple clients.
Calling join(), the server will wait until the client finishes it's work